### DISCUSSION TITLE
tutorials/intro-dgs/05-apollo-explorer

### DISCUSSION THREAD

**Discussion by giscus** [apollographql-education/content-feedback#143] [course feedback]
# tutorials/intro-dgs/05-apollo-explorer

https://www.apollographql.com/tutorials/intro-dgs/05-apollo-explorer

**Comment by kendallshearman** [comment]
So my server is running but I'm unable to run a successful query for id with: 

query Query {
  featuredPlaylists {
    id
  }
}

I'm getting the non-null type error mentioned above. I had a look at the final branch and I don't yet have a directory in my project entitled datasources. Is that where I've gone wrong?

**Reply by mabuyo** [reply]
Hey @kendallshearman ! It should work without the data sources, because we've hardcoded the playlist information that will be returned. Have you tried the troubleshooting steps?

> 1. Double check that you've saved the changes made to the `schema.graphqls` and `PlaylistDataFetcher` files.
> 2. Ensure that the `PlaylistDataFetcher` file is annotated with `@DgsComponent`. (If not, it won't be picked up as relevant to the DGS side of things!)
> 3. Restart the server.

**Reply by kendallshearman** [reply]
Hi @mabuyo ! Ya, I did try those, but I haven't been able to work it out. I'm going to start from scratch and try again because it's totally possible I missed something. Will let you know if I get stuck a second time.
